nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/services/azure/storage/AzureDataLakeStorageFileResourceService.java

```diff
@@ -0,0 +1,151 @@
+        configuration = """
+                "Filesystem Name" = "${azure.filesystem}"
+                "Directory Name" = "${azure.directory}"
+                "File Name" = "${azure.filename}"
+
+                The "ADLS Credentials" property should specify an instance of the ADLSCredentialsService in order to provide credentials for accessing the filesystem.
+                """
+)
+public class AzureDataLakeStorageFileResourceService extends AbstractControllerService implements FileResourceService {
+
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            ADLS_CREDENTIALS_SERVICE,
+            FILESYSTEM,
+            DIRECTORY,
+            FILE
+    );
+
+    private volatile DataLakeServiceClientFactory clientFactory;
+    private volatile ConfigurationContext context;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        this.clientFactory = new DataLakeServiceClientFactory(getLogger(), getProxyOptions(context));
+        this.context = context;
+    }
+
+    @OnDisabled
+    public void onDisabled() {
+        this.clientFactory = null;
+        this.context = null;
+    }
+
+    @Override
+    public FileResource getFileResource(Map<String, String> attributes) {
+        DataLakeServiceClient client = getStorageClient(attributes);
+        try {
+            return fetchBlob(client, attributes);
+        } catch (final DataLakeStorageException | IOException e) {
+            throw new ProcessException("Failed to fetch file from ADLS Storage", e);
+        }
+    }
+
+    protected DataLakeServiceClient getStorageClient(Map<String, String> attributes) {
+        final ADLSCredentialsService credentialsService = context.getProperty(ADLS_CREDENTIALS_SERVICE)
+                .asControllerService(ADLSCredentialsService.class);
+        return clientFactory.getStorageClient(credentialsService.getCredentialsDetails(attributes));
+    }
+
+    /**
+     * Fetching file from the provided filesystem and directory in ADLS.
+     *
+     * @param storageClient azure data lake service client
+     * @param attributes configuration attributes
+     * @return fetched file as FileResource
+     * @throws IOException exception caused by missing parameters or blob not found
+     */
+    private FileResource fetchBlob(final DataLakeServiceClient storageClient, final Map<String, String> attributes) throws IOException {
+        final String fileSystem = context.getProperty(FILESYSTEM).evaluateAttributeExpressions(attributes).getValue();
+        final String directory = context.getProperty(DIRECTORY).evaluateAttributeExpressions(attributes).getValue();
+        final String file = context.getProperty(FILE).evaluateAttributeExpressions(attributes).getValue();
+
+        if (isBlank(fileSystem) || isBlank(file)) {
```

Comment:
There are already methods provided in `AbstractAzureDataLakeStorageProcessor` to check the `fileSystem` and `file` properties, so these could be used here too. These are `evaluateFileSystemProperty` and `evaluateFileNameProperty` methods.